src/App.php

```diff
@@ -1099,11 +1111,11 @@ protected function outputResponse(string $data, array $headers): void
 
         if ($lateError !== null) {
             echo "\n" . '!! FATAL UI ERROR: ' . $lateError . ' !!' . "\n";
-
+            $this->callBeforeExit(); // output before forced exit TODO exit here is really needed?
```

Comment:
see above what is "lateError", basically a fatal error we cannot recover from

the added line is wrong, we must simply echo the fatal error and exit with a non-zero code (as already coded)

Reply:
OK